src/base.ts

```diff
@@ -11,4 +38,93 @@ export default abstract class extends Command {
       }
     }
   }
+
+  async recordActionFinished(action: string, metadata: MetricMetadata = {}, rawDocument?: string) {
+    if (rawDocument !== undefined) {
+      try {
+        const {document} = await this.parser.parse(rawDocument);
+        if (document !== undefined) {
+          metadata = MetadataFromDocument(document, metadata);
+          metadata['source'] = this.generateSHA256(document.info().title());
```

Comment:
Why sha256? Really this is so expensive!

I don't want to be intrusive but md5 is way faster than sha1, obviously faster than sha256 and for this I think it will be enough.

Also, using only the title is not enough. Tons of title will collide. I instead suggest you to use a combination of Title, description, version, etc.

Reply:
I thought that was some requirement from NewRelic, but I didn't read carefully that it was just a SonarCloud recommendation.

Even the `combination of Title, description, version, etc` will collide as the `new file` command generates the same file content.

For reference, this will be used in calculating [Time to first API Design](https://github.com/Amzani/asyncapi-metrics-dashboard/issues/2#issuecomment-1946098809)

```sql
SELECT filter(latest(endTimestamp),
                      WHERE action = 'generate:fromTemplate'
                      AND source is not null
                      AND success = true)/1000 - 
              filter(latest(endTimestamp),
                      WHERE action = 'new:file'
                      AND source is not null
                      AND success = true)/1000 AS 'TS'
FROM Metric FACET source, user
```

After revisiting this query, the source will change any way between file creation and generation. 

I recommend disabling collecting the source for now and solving this in a new PR.
cc @peter-rr , @smoya, @KhudaDad414

Reply:
Agree 👍 Let me disable the source collection then and open a new issue about it.

Reply:
Source collection already disabled, and new issue related to the topic has been opened: https://github.com/asyncapi/cli/issues/1225